src/components/pages/RiskyTrovesPage.tsx

```diff
@@ -1,34 +1,159 @@
+                    startingAt: clampedPage * pageSize,
+                });
+
+                let debtInFront = previousDebt;
+
+                const enriched: TroveWithDebtInFront[] = currentPage.map(
+                    trove => {
+                        const enrichedTrove = Object.assign(
+                            Object.create(Object.getPrototypeOf(trove)),
+                            trove,
+                            {
+                                debtInFront,
+                            }
+                        ) as TroveWithDebtInFront;
+
+                        debtInFront = debtInFront.add(trove.debt);
+                        return enrichedTrove;
+                    }
+                );
+
+                if (mounted) {
+                    setTroves(enriched);
+                }
+            } finally {
+                setLoading(false);
+            }
+        };
+
+        fetchRiskyTroves();
+
+        return () => {
+            mounted = false;
+        };
+    }, [sfStablecoin, clampedPage, pageSize, reload]);
+
+    useEffect(() => {
+        forceReload();
+    }, [forceReload, numberOfTroves]);
+
+    const handlePrevious = () => {
+        if (currentPage > 1) {
+            setCurrentPage(currentPage - 1);
+        }
+    };
+
+    const handleNext = () => {
+        if (currentPage < totalPages) {
+            setCurrentPage(currentPage + 1);
+        }
+    };
+
+    return (
+        <div className='flex w-full flex-col'>
+            <main className='container mx-auto flex-grow px-2 py-5'>
+                <h1 className='mb-4 font-primary text-6 font-semibold'>
+                    Risky Troves
+                </h1>
+                <p className='mb-2 max-w-3xl text-[#565656]'>
+                    Track and liquidate risky Troves to maintain protocol
+                    stability and earn rewards. Troves with a collateral ratio
+                    below 110% (or 150% in Recovery Mode) are at risk, meaning
+                    their FIL collateral may not fully cover their debt.
+                </p>
+
+                <div className='relative flex min-h-[300px] items-center justify-center overflow-hidden'>
+                    {loading ? (
+                        <div className='flex h-[300px] items-center justify-center'>
+                            <Spinner />
+                        </div>
+                    ) : (
+                        <CoreTable
+                            troves={troves}
+                            price={price}
+                            isConnected={isConnected}
+                            sfStablecoin={sfStablecoin}
+                            currentPage={currentPage}
+                            totalPages={totalPages}
+                            onNext={handleNext}
+                            onPrevious={handlePrevious}
+                        />
+                    )}
+                </div>
```

Comment:
Every pagination action shows this Spinner component and disable entire table.
This is not good in terms of UX.
Showing the Spinner component should happen in the table instead.

Reply:
This v1 overlay style is the best UI.
<img width="1664" height="930" alt="Fullscreen_2025_08_12_12_43" src="https://github.com/user-attachments/assets/35aaaa55-b474-4a02-9450-825c8c9bc51b" />